### PR TITLE
Merge upstream v1.12.2

### PR DESCRIPTION
**Pull request by tomtseng**
(This PR should not be squashed.)

### CODE REVIEW

File: cpp/tests/results/runNNCanaryTests.txt
Comment (ed1d1a8d) [inline cpp/tests/results/runNNCanaryTests.txt:1]: Is it okay that this file reports a conflict on a 3-way merge?

File: cpp/tests/results/runNNSymmetriesTestFP16.txt
Comment (ed1d1a8d) [inline cpp/tests/results/runNNSymmetriesTestFP16.txt:1]: Likewise, is it okay that this file reports a conflict on a 3-way merge?

File: cpp/tests/results/runNNSymmetriesTest.txt
Comment (ed1d1a8d) [inline cpp/tests/results/runNNSymmetriesTest.txt:1]: Likewise, is it okay that this file reports a conflict on a 3-way merge?

File: cpp/tests/results/runNNOnTinyBoardTest.txt
Comment (ed1d1a8d) [inline cpp/tests/results/runNNOnTinyBoardTest.txt:1]: Likewise, is it okay that this file reports a conflict on a 3-way merge?

**Review by ed1d1a8d — COMMENTED** [review]
Just had a few questions on some test logs. Everything else looks okay and had no 3-way merge-conflicts.

**Comment by tomtseng** [comment]
I eyeballed the changes in the conflicting test files and I only saw changes in rounding. Hard to say with full confidence, though, because admittedly I don't look at these long test files with full diligence

**Comment by tomtseng** [comment]
For the conflicting test files I think I just reran the tests and committed the results. Eyeballing the diffs I think everything looks OK, just some changes in rounding

**Comment by tomtseng** [comment]
some of these conflicts are due to us adding fields like `selectionProb` to analysis JSON, will rerun tests without these added fields today and look at the diff of the listed files again

**Review by ed1d1a8d — APPROVED** [review]
On closer inspection of the 3-way merge conflicts, the conflict is just due to a change in floating point number precision in printing. LGTM!

**Pull request merged**